Functions to convert values between host and big-/little-endian byte order has
been added in glibc 2.9, and RHEL5 / CentOS5 have glibc 2.5.
So reimplement a part of them if glibc < 2.9

diff -Naur burp-2.0.38/src/burp.h.ORIG burp-2.0.38/src/burp.h
--- burp-2.0.38/src/burp.h.ORIG	2016-05-01 07:59:47.000000000 +0200
+++ burp-2.0.38/src/burp.h	2016-05-25 23:27:55.448347482 +0200
@@ -140,4 +140,22 @@
 	#define __PDP_ENDIAN	PDP_ENDIAN
 #endif
 
+#if !defined(htobe64) && defined(__GLIBC__) && __GLIBC__ <= 2 && __GLIBC_MINOR__ < 9
+	#include <sys/param.h>
+	#include <byteswap.h>
+	#if __BYTE_ORDER == __LITTLE_ENDIAN
+		#define htobe64(x) bswap_64 (x)
+		#define htole64(x) (x)
+		#define be64toh(x) bswap_64 (x)
+		#define le64toh(x) (x)
+	#elif __BYTE_ORDER == __BIG_ENDIAN
+		#define htobe64(x) (x)
+		#define htole64(x) bswap_64 (x)
+		#define be64toh(x) (x)
+		#define le64toh(x) bswap_64 (x)
+	#else
+		#error byte order not supported
+	#endif
+#endif
+
 #endif
